packages/calling-stateful-client/src/BreakoutRoomsSubscriber.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* @conditional-compile-remove(breakout-rooms) */
+import {
+  BreakoutRoom,
+  BreakoutRoomsCallFeature,
+  BreakoutRoomsEventData,
+  BreakoutRoomsSettings
+} from '@azure/communication-calling';
+/* @conditional-compile-remove(breakout-rooms) */
+import { CallContext } from './CallContext';
+/* @conditional-compile-remove(breakout-rooms) */
+import { CallIdRef } from './CallIdRef';
+
+/* @conditional-compile-remove(breakout-rooms) */
+/**
+ * @private
+ */
+export class BreakoutRoomsSubscriber {
+  private _callIdRef: CallIdRef;
+  private _context: CallContext;
+  private _breakoutRoomsFeature: BreakoutRoomsCallFeature;
+
+  constructor(callIdRef: CallIdRef, context: CallContext, breakoutRoomsFeature: BreakoutRoomsCallFeature) {
+    this._callIdRef = callIdRef;
+    this._context = context;
+    this._breakoutRoomsFeature = breakoutRoomsFeature;
+
+    this.subscribe();
+  }
+
+  public unsubscribe = (): void => {};
+
+  private subscribe(): void {
+    this._breakoutRoomsFeature.on('breakoutRoomsUpdated', this.onBreakoutRoomsUpdated);
+  }
+
+  private onBreakoutRoomsUpdated = (eventData: BreakoutRoomsEventData): void => {
+    if (eventData.type === 'assignedBreakoutRooms') {
+      this.onAssignedBreakoutRoomUpdated(eventData.data as BreakoutRoom);
```

Comment:
Don't use `as` - this breaks type safety.
For example, `event.data` could be `undefined` but the `as` removes the typecheck guard for checking against undefined.

Instead this should word
```ts
    if (!eventData.data) {
      return;
    }

    if (eventData.type === 'assignedBreakoutRooms' && eventData.data) {
      this.onAssignedBreakoutRoomUpdated(eventData.data);
    } else if (eventData.type === 'breakoutRoomsSettings') {
      this.onBreakoutRoomSettingsUpdated(eventData.data);
    }
```

Reply:
Just a general question about the suggested code as I have played with this kind of thing in the past. How do we know how to pull the type? is it just the property name of the property in the object? because you can't use this kind of thing on the whole object right? you have to go digging to find out?

As a suggestion to @mgamis-msft maybe you can write a little function to use in the check and stick it in the `TypeGuards.ts` file in this package to simplify the code as well